apps/traincascade/cascadeclassifier.cpp

```diff
@@ -340,7 +340,7 @@ int CvCascadeClassifier::fillPassedSamples( int first, int count, bool isPositiv
             if( predict( i ) == 1 )
             {
                 getcount++;
-                printf("%s current samples: %d\r", isPositive ? "POS":"NEG", getcount);
+                std::cout << (isPositive ? "POS" : "NEG") << " current samples: " << getcount << "\r" << std::flush;
```

Comment:
Consider using this:
```
printf(...); fflush(stdout);
```

Reply:
@alalek, I have no opportunity to test that C-style version now. Would you mind keeping C++ version here?
P.S.: thanks for resurrecting this change, I completely forgot about it.

Reply:
BTW, Main concern here is to avoid changing of unrelated code (`printf()` statement).

These lines should do the same things:
```
fflush(stdout);
std::cout.flush();
std::cout << std::flush;
```

`fflush(stdout);` has been selected because it correlates with `printf()` (both are C-style).

Reply:
I've reverted the change and use fflush as you suggested.